src/NuGet.Core/NuGet.LibraryModel/CentralPackageVersionNameComparer.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright(c) .NET Foundation.All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace NuGet.LibraryModel
+{
+    public sealed class CentralPackageVersionNameComparer : IEqualityComparer<CentralPackageVersion>
+    {
+        private static readonly Lazy<CentralPackageVersionNameComparer> _defaultComparer = new Lazy<CentralPackageVersionNameComparer>(() => new CentralPackageVersionNameComparer());
+
+        public static CentralPackageVersionNameComparer Default
+        {
+            get
+            {
+                return _defaultComparer.Value;
+            }
+        }
+
+        private CentralPackageVersionNameComparer()
+        {
```

Comment:
Maybe a comment would be useful to explain why this is disabled, and why a Singleton is important to be used here?

Reply:
I added some comments.